src/openeo_udf/api/datacube.py

```diff
@@ -277,6 +279,179 @@ def from_data_collection(data_collection: 'openeo_udf.server.data_model.data_col
+        if fmt.lower()=='netcdf':
+            return DataCube._save_DataArray_to_NetCDF(filename, self.get_array())
+        if fmt.lower()=='json':
+            return DataCube._save_DataArray_to_JSON(filename, self.get_array())
+    
+    
+    @staticmethod
+    def _load_DataArray_from_JSON(filename) -> xarray.DataArray:
+        with open(filename) as f:
+            # get the deserialized json
+            d=json.load(f)
+            d['data']=numpy.array(d['data'],dtype=numpy.dtype(d['attrs']['dtype']))
+            for k,v in d['coords'].items():
+                # prepare coordinate 
+                d['coords'][k]['data']=numpy.array(v['data'],dtype=v['attrs']['dtype'])
+                # remove dtype and shape, because that is included for helping the user
+                if d['coords'][k].get('attrs',None) is not None:
+                    d['coords'][k]['attrs'].pop('dtype',None)
+                    d['coords'][k]['attrs'].pop('shape',None)
+            
+            # remove dtype and shape, because that is included for helping the user
+            if d.get('attrs',None) is not None:
+                d['attrs'].pop('dtype',None)
+                d['attrs'].pop('shape',None)
+            # vonvert to xarray
+            r=xarray.DataArray.from_dict(d)
+            del d
+        # build dimension list in proper order
+        dims=list(filter(lambda i: i!='t' and i!='bands' and i!='x' and i!='y',r.dims))
+        if 't' in r.dims: dims+=['t']
+        if 'bands' in r.dims: dims+=['bands']
+        if 'x' in r.dims: dims+=['x']
+        if 'y' in r.dims: dims+=['y']
+        # return the resulting data array
+        return r.transpose(*dims)
+    
+    
+    @staticmethod
+    def _load_DataArray_from_NetCDF(filename) -> xarray.DataArray:
+        # load the dataset and convert to data array
+        ds=xarray.open_dataset(filename, engine='h5netcdf')
+        r=ds.to_array(dim='bands')
+        # build dimension list in proper order
+        dims=list(filter(lambda i: i!='t' and i!='bands' and i!='x' and i!='y',r.dims))
+        if 't' in r.dims: dims+=['t']
+        if 'bands' in r.dims: dims+=['bands']
+        if 'x' in r.dims: dims+=['x']
+        if 'y' in r.dims: dims+=['y']
+        # return the resulting data array
+        return r.transpose(*dims)
+    
+    
+    @staticmethod
+    def _save_DataArray_to_JSON(filename, array: xarray.DataArray):
+        # to deserialized json
+        jsonarray=array.to_dict()
+        # add attributes that needed for re-creating xarray from json
+        jsonarray['attrs']['dtype']=str(array.values.dtype)
+        jsonarray['attrs']['shape']=list(array.values.shape)
+        for i in array.coords.values():
+            jsonarray['coords'][i.name]['attrs']['dtype']=str(i.dtype)
+            jsonarray['coords'][i.name]['attrs']['shape']=list(i.shape)
+        # custom print so resulting json file is humanly easy to read
+        with open(filename,'w') as f:
```

Comment:
Any reason why json.dumps(mydict,indent=2) is not good enough?

Reply:
For two reasons:
- this way it is easy to obtain a quick overview just by grepping in a bash terminal
- indent would put every number in a separate line + indentation spaces -> the file size would explode